Apps/WebClient/src/ClientApp/src/components/vaccinationStatusResult.vue

```diff
@@ -63,11 +79,40 @@ export default class VaccinationStatusResultView extends Vue {
         }
         return new DateWrapper(date).format(DateWrapper.defaultFormat);
     }
+
+    private download() {
+        this.isDownloading = true;
+        this.getReport({
+            phn: this.status?.personalhealthnumber || "",
+            dateOfBirth: new DateWrapper(this.status?.birthdate || "").format(
+                "yyyy-MM-dd"
+            ),
+        })
+            .then((documentResult) => {
+                if (documentResult) {
+                    const downloadLink = `data:application/pdf;base64,${documentResult.data}`;
+                    fetch(downloadLink).then((res) => {
+                        res.blob().then((blob) => {
+                            saveAs(blob, documentResult.fileName);
+                        });
+                    });
+                }
+            })
+            .catch((err) => {
+                this.logger.error(
+                    `Error retrieving vaccination status pdf: ${err}`
+                );
+            })
+            .finally(() => {
+                this.isDownloading = false;
+            });
+    }
 }
 </script>
 
 <template>
     <div class="d-flex flex-column flex-grow-1">
+        <LoadingComponent :is-loading="isDownloading"></LoadingComponent>
```

Comment:
You could potentially remove this and take advantage of the LoadingComponent on the parent component if your mutation set the state.status to LoadStatus.REQUESTED.

Reply:
It looks like you're not using mutations, so that might be more trouble than it's worth.